### PR TITLE
Markup

### PR DESCRIPTION
**Pull request by Alanamiael**

### CODE REVIEW

File: src/components/MainTableBanner/MainTableBanner.module.scss [hunk 1]
```diff
@@ -1,8 +1,8 @@
 @import '../../styles/variables';
-@import url(https://fonts.googleapis.com/css?family=Open+Sans:300,regular,500,600,700,800,300italic,italic,500italic,600italic,700italic,800italic);
-@import url(https://fonts.googleapis.com/css?family=Inter:100,200,300,regular,500,600,700,800,900);
+
 .flex {
```
Comment (Yarolaw) [inline src/components/MainTableBanner/MainTableBanner.module.scss]: incorrect name for class

File: src/components/MainTableHeadings/index.tsx [hunk 2]
```diff
@@ -0,0 +1,15 @@
+import cl from './MainTableHeadings.module.scss';
+
+const MainTableHeadings = () => {
+  return (
+    <div className={cl.pStyleForHeadings}>
```
Comment (Yarolaw) [inline src/components/MainTableHeadings/index.tsx]: the same here (incorrect name)

File: src/components/MainTableHeadings/index.tsx [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+      <p>Temp, °C</p>
```
Comment (Yarolaw) [inline src/components/MainTableHeadings/index.tsx]: here we can use map for render the <p> tag

File: src/components/WeekWeather/index.tsx [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+import cl from './WeekWeather.module.scss';
+
+interface WeekWeatherProps {
+  data: any;
+}
+
+const WeekWeather: FC<WeekWeatherProps> = ({ data }) => {
+  return (
+    <div className={cl.flex}>
+      <DayWeather data={data?.forecast?.forecastday[1]} />
```
Comment (Yarolaw) [inline src/components/WeekWeather/index.tsx]: the same here, better to use map

File: src/styles/App.module.scss [hunk 5]
```diff
@@ -0,0 +1,10 @@
+.app {
+  display: flex;
```
Comment (Yarolaw) [inline src/styles/App.module.scss]: Where do you see styles which repeat you can use mixins for that

File: src/styles/index.scss [hunk 6]
```diff
@@ -1,3 +1,12 @@
...
+@import url(https://fonts.googleapis.com/css?family=Inter:100,200,300,regular,500,600,700,800,900);
+
 * {
+  padding: 0;
+  margin: 0;
+  border: 0;
+}
+
+body {
   margin: 30px auto;
```
Comment (Yarolaw) [inline src/styles/index.scss]: Better set in the body tag styles for common fonts to avoid doing it directly

File: src/App.tsx [hunk 7]
```diff
@@ -1,13 +1,59 @@
-import './styles/App.scss';
+import { useEffect, useState } from 'react';
+import axios from 'axios';
```
Comment (Yarolaw) [inline src/App.tsx]: All API operations should be moved from component to services folder

File: src/App.tsx [hunk 8]
```diff
@@ -1,13 +1,59 @@
...
+import MeterReadings from './components/MeterReadings';
+import WeekWeather from './components/WeekWeather';
+import cl from './styles/App.module.scss';
 
 function App() {
+  const [data, setData] = useState();
+  const [location, setLocation] = useState('Seoul');
+  const [todayWeather, setTodayWeather] = useState(null);
+
+  const url = `http://api.weatherapi.com/v1/forecast.json?key=17793347648846f1a1d115955230611&q=${location}&days=7&aqi=no&alerts=no`;
```
Comment (Yarolaw) [inline src/App.tsx]: Should be moved to services

File: src/App.tsx [hunk 9]
```diff
@@ -1,13 +1,59 @@
...
+
+  useEffect(() => {
+    const fetchData = async () => {
```
Comment (Yarolaw) [inline src/App.tsx]: also here

File: src/App.tsx [hunk 10]
```diff
@@ -1,13 +1,59 @@
...
+          ),
+        });
+      } catch (error) {
+        console.log(error);
+      }
+    };
+    fetchData();
+  }, []);
+
+  useEffect(() => {}, []);
```
Comment (Yarolaw) [inline src/App.tsx]: unnecessary useEffect

File: src/components/DayWeather/DayWeather.module.scss [hunk 11]
```diff
@@ -0,0 +1,55 @@
+@import url(https://fonts.googleapis.com/css?family=Open+Sans:300,regular,500,600,700,800,300italic,italic,500italic,600italic,700italic,800italic);
+@import url(https://fonts.googleapis.com/css?family=Inter:100,200,300,regular,500,600,700,800,900);
```
Comment (Yarolaw) [inline src/components/DayWeather/DayWeather.module.scss]: Should be moved to index.css

File: src/components/Heading/Heading.module.scss [hunk 12]
```diff
@@ -14,3 +15,55 @@
     line-height: normal;
   }
 }
+
+.myInput {
```
Comment (Yarolaw) [inline src/components/Heading/Heading.module.scss]: incorrect name

File: src/components/Heading/index.tsx [hunk 13]
```diff
@@ -1,12 +1,58 @@
-import SearchInput from '../SearchInput';
-
+import { FC } from 'react';
+import axios from 'axios';
```
Comment (Yarolaw) [inline src/components/Heading/index.tsx]: Api operations should be moved

File: src/components/Heading/index.tsx [hunk 14]
```diff
@@ -1,12 +1,58 @@
...
+}
+
+const Heading: FC<HeadingProps> = ({
+  setData,
+  location,
+  setLocation,
+  setTodayWeather,
+  url,
+}) => {
+  const searchLocation = () => {
```
Comment (Yarolaw) [inline src/components/Heading/index.tsx:31]: To the services

File: src/components/MainContent/index.tsx [hunk 15]
```diff
@@ -0,0 +1,24 @@
+import MainContentElement from '../MainContentElement';
+import MainTableHeadings from '../MainTableHeadings';
+import cl from './MainContent.module.scss';
+
+const MainContent = ({ todayWeatherHours }: { todayWeatherHours: any }) => {
+  return (
+    <div className={cl.flex}>
+      <MainTableHeadings />
+      {todayWeatherHours?.map((item: unknown, index: number) => {
```
Comment (Yarolaw) [inline src/components/MainContent/index.tsx:9]: Why item: unknown ?

File: src/components/MainContentElement/MainContentElement.module.scss [hunk 16]
```diff
@@ -0,0 +1,46 @@
+@import url(https://fonts.googleapis.com/css?family=Open+Sans:300,regular,500,600,700,800,300italic,italic,500italic,600italic,700italic,800italic);
```
Comment (Yarolaw) [inline src/components/MainContentElement/MainContentElement.module.scss]: Move to the index.css

File: src/components/MainContentElement/index.tsx [hunk 17]
```diff
@@ -0,0 +1,51 @@
...
+        <div className={cl.flexColumn}>
+          <p className={cl.time}>
+            {dayjs(hoursWeather[0].time).format('H:mm')}
+          </p>
+          <img
+            src={hoursWeather[0].condition.icon}
+            alt={hoursWeather[0].condition.text}
+          />
+          <div className={cl.flexColumn}>
+            <p className={cl.info}>{hoursWeather[0].temp_c}</p>
```
Comment (Yarolaw) [inline src/components/MainContentElement/index.tsx]: Better to use map here

File: src/components/MainContentElement/index.tsx [hunk 18]
```diff
@@ -0,0 +1,51 @@
...
+        <div className={cl.flexColumn}>
+          <p className={cl.time}>
+            {dayjs(hoursWeather[1].time).format('H:mm')}
+          </p>
+          <img
+            src={hoursWeather[1].condition.icon}
+            alt={hoursWeather[1].condition.text}
+          />
+          <div className={cl.flexColumn}>
+            <p className={cl.info}>{hoursWeather[1].temp_c}</p>
```
Comment (Yarolaw) [inline src/components/MainContentElement/index.tsx]: Also here, use map

**Comment by Alanamiael** [comment]
I wanna create new one

**Pull request closed without merging**